gocd/templates/gocd-agent-controller.yaml

```diff
@@ -82,6 +95,9 @@ spec:
           resources:
 {{ toYaml .Values.agent.resources | indent 12 }}
           env:
+            {{- with .Values.agent.env.extraEnvVars }}
+{{ toYaml . | indent 12 }}
+            {{- end }}
```

Comment:
Why is this being moved? This changes and potentially breaks semantics if folks are trying to override values directly rather than using values.

Reply:
Right, I forgot to mention it, it allows me to do:
```yaml
agent:
  env:
    agentAutoRegisterHostname: "myname-$(POD_INDEX)"
    extraEnvVars:
      - name: POD_INDEX
        valueFrom:
          fieldRef:
            fieldPath: metadata.labels['apps.kubernetes.io/pod-index']
```
so that my agents get registered as `myname-0`, `myname-1`, ...

It is technically breaking if you want to do env subst the opposite way I want to, so in alternative I can always add that `POD_INDEX` variable at the top of the list if I'm deploying a StatefulSet and put the order back.

Reply:
Same result can be obtained by creating `AGENT_AUTO_REGISTER_HOSTNAME` manually, I reverted this.